src/compiler/checker.ts

```diff
@@ -1355,15 +1355,34 @@ namespace ts {
             }
 
             const isRelative = isExternalModuleNameRelative(moduleName);
+            let quotedName: string | undefined;
             if (!isRelative) {
-                const symbol = getSymbol(globals, '"' + moduleName + '"', SymbolFlags.ValueModule);
+                quotedName = '"' + moduleName + '"';
+                const symbol = getSymbol(globals, quotedName, SymbolFlags.ValueModule);
                 if (symbol) {
                     // merged symbol is module declaration symbol combined with all augmentations
                     return getMergedSymbol(symbol);
                 }
             }
 
             const resolvedModule = getResolvedModule(getSourceFileOfNode(location), moduleReference);
+            if (resolvedModule && resolvedModule.isUntyped) {
+                Debug.assert(!isRelative); // Only global modules may be untyped
+                Debug.assert(quotedName !== undefined);
+                if (compilerOptions.noImplicitAny) {
+                    error(errorNode,
+                        Diagnostics.A_package_for_0_was_found_at_1_but_is_untyped_Since_noImplicitAny_is_enabled_this_package_must_have_a_declaration,
+                        moduleReference,
+                        resolvedModule.resolvedFileName);
+                }
+
+                const newSymbol = createSymbol(SymbolFlags.ValueModule, quotedName);

```

Comment:
If you must, i would create a singleton `UnkownModuleSymbol` and check for that in `isUntypedModuleSymbol`